domain-mapping/classes/class-dm-media.php

```diff
@@ -0,0 +1,393 @@
+<?php
+/**
+ * Class DM_Media
```

Comment:
The class header does not say what the class does.

On an aside, you could use a multidimensional array `$this->domains[DM_DOMAIN_TYPE_MAIN][$primary, $secondary]` and simplify this class. The advantage would be that this class becomes domain type aware, rather than the responsibility being on you to remember if logic applies to all or some domain types.